### ISSUE TITLE
Overflowing literal lint is silenced in range literals in loops on nightly

### ISSUE THREAD

**Issue by varkor** [rust-lang/rust#60459]
```rust
fn main() {
    for _ in 0..256u8 {} // should warn
}
```
since https://github.com/rust-lang/rust/pull/60330.

**Comment by varkor** [comment]
It *is* hitting the `err.emit()` here:
https://github.com/rust-lang/rust/blob/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_lint/types.rs#L102
So I'm not quite sure why the error isn't being emitted yet.

**Comment by ehuss** [comment]
It appears to be hitting this line of code which is cancelling the diagnostic:

https://github.com/rust-lang/rust/blob/d15fc173818ccea6f8d97af13a30d454f9e4b35c/src/librustc/lint/mod.rs#L746

**Comment by pnkfelix** [comment]
this appears to be a stable-to-beta regression ([playpen](https://play.rust-lang.org/?version=beta&mode=debug&edition=2018&gist=305ac6d5cc504488d6a79f3f6fb69d75))

**Comment by pnkfelix** [comment]
triage: P-high. Assigning to self.

**Comment by estebank** [comment]
We need to change https://github.com/rust-lang/rust/blob/d15fc173818ccea6f8d97af13a30d454f9e4b35c/src/librustc/lint/mod.rs#L871 to not consider desugarings coming from `for` loops as external. If that has regressions in our test suite, then we can add a new desugaring kind to differentiate between different parts of the for loop desugaring and target it further, but I do not expect that to be needed.

**Comment by varkor** [comment]
@pnkfelix: I've opened https://github.com/rust-lang/rust/pull/61098 with a fix, based on @estebank's comment.
